src/training/main.py

```diff
@@ -273,7 +275,16 @@ def main(args):
     scheduler = None
     if 'train' in data and optimizer is not None:
         total_steps = (data["train"].dataloader.num_batches // args.accum_freq) * args.epochs
-        scheduler = cosine_lr(optimizer, args.lr, args.warmup, total_steps)
+        if args.lr_scheduler == "cosine":
+            scheduler = cosine_lr(optimizer, args.lr, args.warmup, total_steps)
+        elif args.lr_scheduler == "const":
+            scheduler = const_lr(optimizer, args.lr, args.warmup, total_steps)
+        elif args.lr_scheduler == "const-cooldown":
+            assert args.cooldown_epochs is not None, "Please specify the number of cooldown epochs for this lr schedule."
+            cooldown_steps = (data["train"].dataloader.num_batches // args.accum_freq) * args.cooldown_epochs
+            scheduler = const_lr_cooldown(optimizer, args.lr, args.warmup, cooldown_steps, args.power_lr, args.end_lr, total_steps)
+        else:
+            print(f"Unknown scheduler, {args.lr_scheduler}")
```

Comment:
I would suggest to exit with an error - as default as cosine, people should get termination as most probably they were providing a name that is not in the schedule list and it will probably be confusing if that leads to execution of cosine schedule - which was not provided in this scenario as  a choice.